### PR TITLE
build(datasets): Release 4.1.0

### PR DESCRIPTION
**Pull request by ElenaKhaustova**
## Description
Release 4.1.0 🥳

Closes https://github.com/kedro-org/kedro-plugins/issues/778

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro-datasets/kedro_datasets/__init__.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """``kedro_datasets`` is where you can find all of Kedro's data connectors."""
 
 __all__ = ["KedroDeprecationWarning"]
-__version__ = "4.0.0"
+__version__ = "4.0.1"
```
Comment (merelcht) [inline kedro-datasets/kedro_datasets/__init__.py:4]: This should be `4.1.0` right?

Comment (ElenaKhaustova) [reply kedro-datasets/kedro_datasets/__init__.py]: ofc, thank you

**Review by ankatiyar — APPROVED** [review]
👍🏾

**Review by merelcht — APPROVED** [review]
:shipit: 🎉

**Review by SajidAlamQB — APPROVED** [review]

File: kedro-datasets/RELEASE.md [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-* Improved PartitionedDataset representation when printing.
 
 ## Bug fixes and other changes
 ## Breaking Changes
 ## Community contributions
 
-# Release 4.0.1
+# Release 4.1.0
 ## Major features and improvements
+* Improved PartitionedDataset representation when printing.
```
Comment (noklam) [inline kedro-datasets/RELEASE.md:10]:
```suggestion
* Improved `partitions.PartitionedDataset` representation when printing.
```

**Review by DimedS — APPROVED** [review]
🚀

**Comment by ElenaKhaustova** [comment]
Fix of docs issue: https://github.com/kedro-org/kedro-plugins/pull/797

**Comment by deepyaman** [comment]
> Fix of docs issue: #797

https://github.com/kedro-org/kedro-plugins/pull/797 is ready to be reviewed/merged. I reverted the rename from `_meth` to `meth`, as that needs to go into a major release (will raise a separate PR on that later).

**Pull request merged**